lib/application/view-models/home/collections_vm.dart

```diff
@@ -13,104 +19,125 @@ abstract class CollectionsVM extends StateNotifier<CollectionsState> {
-  ),
-];
-
 class CollectionsVMImpl extends CollectionsVM {
-  CollectionsVMImpl() : super(LoadingCollectionsState(availableSegments.first)) {
-    _loadCollections();
+  CollectionsVMImpl(this._services) : super(LoadingCollectionsState(availableSegments.first)) {
+    _loadInitialCollections();
   }
 
+  final CollectionServices _services;
+  List<StreamSubscription<CollectionStatus>>? _statusListeners;
+
+  final _cachedCollectionMetadata = <CollectionMetadata>[];
+
   @override
   Future<void> updateCollectionsSegment(CollectionsSegment segment) async {
-    state = LoadingCollectionsState(segment);
-
-    // TODO(matuella): attach logic
-    await Future.delayed(const Duration(seconds: 1), () {});
-
-    final Map<String, List<Collection>> collectionsPerCategory;
-    switch (segment) {
-      case CollectionsSegment.explore:
-        collectionsPerCategory = _mapCollectionsToCategories(mockedCollections);
-        break;
-      case CollectionsSegment.review:
-        collectionsPerCategory =
-            _mapCollectionsToCategories(mockedCollections.where((collection) => !collection.isPristine).toList());
-        break;
+    if (state is LoadingCollectionsState) {
+      state = LoadingCollectionsState(segment);
+      return;
     }
 
-    state = LoadedCollectionsState(collectionsPerCategory, currentSegment: segment);
+    // Filters all the `CollectionMetadata` given the `segment`
+    final filteredMetadata = _cachedCollectionMetadata.where((metadata) {
+      switch (segment) {
+        case CollectionsSegment.explore:
+          return metadata is CompletedCollectionMetadata;
+        case CollectionsSegment.review:
+          return metadata is IncompleteCollectionMetadata;
+      }
+    }).toList();
+
+    final items = _mapMetadataToItems(filteredMetadata);
+    state = LoadedCollectionsState(items, currentSegment: segment);
   }
 
-  Future<void> _loadCollections() async {
-    // TODO(matuella): attach logic
-    await Future.delayed(const Duration(seconds: 1), () {});
+  Future<void> _loadInitialCollections() async {
+    final collections = await _services.getAllCollectionsStatus();
+    _cachedCollectionMetadata.addAll(collections.map(_mapStatusToMetadata));
+    return _addCollectionsListeners();
+  }
 
-    final collectionsPerCategory = _mapCollectionsToCategories(mockedCollections);
-    state = LoadedCollectionsState(collectionsPerCategory, currentSegment: state.currentSegment);
+  Future<void> _addCollectionsListeners() async {
+    final statusStreams = await _services.listenToAllCollectionsStatus();
+    // Listens to each individual stream, as we don't want to update all the objects once a single one changes
+    _statusListeners = statusStreams.map((statusStream) {
+      var isFirstUpdate = true;
+      return statusStream.listen((status) {
+        final cachedIndex = _cachedCollectionMetadata.indexWhere((metadata) => status.collection.id == metadata.id);
+        // TODO(matuella): TEST IF THIS IS NEEDED
```

Comment:
Is this a forgotten TODO or it will be made in the future?

Reply:
Because this PR won't be merge into main directly, I will still address this TODO, but in a future PR, as the amount of work to mock this test right now is not worth it.